### DISCUSSION TITLE
Add documentation of full table instance variables including state for each hook

### DISCUSSION THREAD

**Discussion by CaitlinWeb-st** [TanStack/table#2735] [Ideas]
#### version: 7.5.0

While working on some "advanced" functionality for tables that use `react-table` and looking at the documentation it isn't obvious that you can manipulate the state for the hooks. (Maybe this isn't the case with all of them.)

Specifically for `useSortBy`, the [Instance Properties](https://react-table.tanstack.com/docs/api/useSortBy#instance-properties) are described in documentation as only these things:
* `rows`
* `preSortedRows`
* `toggleSortBy`

There's no mention of:
* `setSortBy`
* `state: { sortBy }`

My use-case involved saving the sort options to localStorage, so I wanted to be able to use or set the sort's state (within a `useEffect`). It wasn't obvious that was possible until I looked through everything returned by `useTable`.

Could the documentation be updated with the additional instance variables that `useTable` includes with each of the hooks?

**Comment by gargroh** [comment]
Feel free to PR for the same :)
